Lib/functools.py

```diff
@@ -193,6 +193,84 @@ def total_ordering(cls):
     return cls
 
 
+################################################################################
+### topological sort
+################################################################################
+
+
+class CycleError(ValueError):
+    def __init__(self, data):
+        # Sort the data just to make the output consistent, for use in
+        # error messages.  That's convenient for doctests.
+        items = ('{!r}:{!r}'.format(key, value) for key, value in sorted(data.items()))
```

Comment:
I find this dictionary format hard to read. Realistically the algorithm will always detect a cycle in form of a cycle, such as ``1 -> 2 -> 3 -> 1``, this feels much more readable to me as opposed to ``{1:{2}, 2:{3}, 3:{1}}``. We should take care that if the cycle is `` 1 -> 2 -> 3 -> 2`` maybe just show `` 2 -> 3 -> 2``.

Reply:
Agreed, will update soon.